src/cpu/flexcpu/inflight_inst.cc

```diff
@@ -470,16 +520,21 @@ InflightInst::setMiscRegOperand(const StaticInst* si, int dst_idx,
 MiscReg
 InflightInst::readMiscReg(int misc_reg)
 {
-    panic("readMiscReg() not implemented!");
-    return 0;
-    // TODO
+    return backingISA->readMiscReg(misc_reg, backingContext);
 }
 
 void
 InflightInst::setMiscReg(int misc_reg, const MiscReg& val)
 {
-    panic("setMiscReg() not implemented!");
-    // TODO
+    // O3 hides multiple writes to the same misc reg during execution, but due
+    // to potential side effects of each access, I don't think we should be
+    // hiding them. Instead, we will replay each write sequentially during
+    // commit time.
```

Comment:
Hmm... I'm not sure this will make any difference. Since you're functionally executing all of the writes on the same tick and there's no timing done for the side effects I think it'll be the same either way. This seems safer and easier to implement, though.

Reply:
I thought all of those things too